src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/DetectorOptions.java

```diff
@@ -69,6 +71,7 @@
      */
     public DetectorOptions setFormat(String format) {
         this.format = format;
+        pattern = null;
```

Comment:
@fabotronix 
Can you explain why we need to assign `null` here? Doesn't it work without this line?